src/Components/Web.Extensions/src/InputFile/InputFileJsCallbacksRelay.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    internal class InputFileJsCallbacksRelay : IDisposable
+    {
+        private readonly IInputFileJsCallbacks _callbacks;
+
+        public IDisposable DotNetReference { get; }
+
+        public InputFileJsCallbacksRelay(IInputFileJsCallbacks callbacks)
+        {
+            _callbacks = callbacks;
+
+            DotNetReference = DotNetObjectReference.Create(this);
+        }
+
+        [JSInvokable]
+        public Task NotifyChange(FileListEntry[] files)
```

Comment:
Does this actually work? I thought that `JSInterop` only exposed `public` methods on `public` types.

If that's the case you could make `InputFileJsCallbacksRelay` public but its constructor internal.

Reply:
It does work the way it is now - if that's a bug that needs to be fixed, I could change `InputFileJsCallbacksRelay` to public and make its constructor internal per your suggestion.

Reply:
Ah, I must have been misremembering the rules. If that already works, we wouldn't "fix" it by stopping it from working because that would be an unnecessary breaking change. What you have looks ideal then.